ci-operator/step-registry/gather/core-dump/gather-core-dump-commands.sh

```diff
@@ -12,5 +12,27 @@ echo "Gathering node core dumps ..."
 mkdir -p ${ARTIFACT_DIR}/core-dumps
 
 oc adm must-gather --dest-dir="${ARTIFACT_DIR}/core-dumps" -- sh -c "/usr/bin/gather_core_dumps || true"
+
+ls -R ${ARTIFACT_DIR}/core-dumps
+find ${ARTIFACT_DIR}/core-dumps -type f
+num_core_dumps=$(find ${ARTIFACT_DIR}/core-dumps/*/ -type f | wc -l)
+
 tar -czC "${ARTIFACT_DIR}/core-dumps" -f "${ARTIFACT_DIR}/core.dumps.tar.gz" .
 rm -rf "${ARTIFACT_DIR}/core-dumps"
+
+# if there are any files collected from the core-dump gather and $FAIL_ON_CORE_DUMP is true then exit 1
+# so the job can see it and be marked as a failure
+if [ "${FAIL_ON_CORE_DUMP}" == "true" ]; then
+  if [ $num_core_dumps -ne 0 ]; then
+    echo "Fail: Found core dump files."
+    mkdir -p ${ARTIFACT_DIR}/junit/
+    echo '<testsuite name="gather core dump" tests="1" failures="1">
+            <testcase name="core files found">
+              <failure message="">Fail: Found core dump files.</failure>
+            </testcase>
+          </testsuite>' > ${ARTIFACT_DIR}/junit/junit_core_dump_status.xml
```

Comment:
@wking , you suggested [here](https://coreos.slack.com/archives/C01CQA76KMX/p1628017582087700?thread_ts=1628015994.084700&cid=C01CQA76KMX) that it would be smart to throw this message in a junit. Nothing fancy, but this seems
to [work](https://search.ci.openshift.org/?search=Fail%3A+Found+core+dump+files&maxAge=336h&context=1&type=junit&name=&excludeName=&maxMatches=5&maxBytes=20971520&groupBy=job). got a better idea I should try?